File: Examples.VB/Features/Shape/AddShapeWithCustomName.vb
Namespace Features.Shape
    Public Class AddShapeWithCustomName
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            Dim worksheet As IWorksheet = workbook.Worksheets(0)
            Dim shape As IShape = worksheet.Shapes.AddShape("custom parallelogram", AutoShapeType.Parallelogram, 1, 1, 200, 100)

            'Get shape by name
            Dim parallelogram As IShape = worksheet.Shapes("custom parallelogram")
            parallelogram.Fill.Color.RGB = System.Drawing.Color.Red
        End Sub
    End Class
End Namespace

File: Examples/Features/Shape/AddShapeWithCustomName.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Shape
{
    public class AddShapeWithCustomName : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            IWorksheet worksheet = workbook.Worksheets[0];
            GrapeCity.Documents.Excel.Drawing.IShape shape = worksheet.Shapes.AddShape("custom parallelogram", GrapeCity.Documents.Excel.Drawing.AutoShapeType.Parallelogram, 1, 1, 200, 100);

            //Get shape by name
            GrapeCity.Documents.Excel.Drawing.IShape parallelogram = worksheet.Shapes["custom parallelogram"];
            parallelogram.Fill.Color.RGB = System.Drawing.Color.Red;
        }

        public override bool IsNew
        {
            get
            {
                return true;
            }
        }

    }
}
